imap/caldav_alarm.c

```diff
@@ -1233,14 +1235,30 @@ EXPORTED int caldav_alarm_process(time_t runtime, time_t *intervalp)
 
     caldav_alarm_close(alarmdb);
 
-    int i;
+    int i, cnt = 0;
+    char *prev_userid = NULL;
     for (i = 0; i < rock.list.count; i++) {
         struct caldav_alarm_data *data = ptrarray_nth(&rock.list, i);
+        char *userid = mboxname_to_userid(data->mboxname);
+
+        if (strcmpnull(userid, prev_userid)) {
+            /* different user - reset counter */
+            free(prev_userid);
+            prev_userid = userid;
+            cnt = 1;
+        }
+        else if (++cnt >= MAX_CONSECUTIVE_ALARMS_PER_USER) {
```

Comment:
If there's only a single user with more than 50 alarms, this will cause an infinite loop!  You need to at least zero out `cnt` again.